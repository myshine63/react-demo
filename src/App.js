import React from 'react';
import './App.css';
import {BrowserRouter, Route, Switch} from "react-router-dom";
import asyncLoad from "./utils/asyncLoad";
//--- page component ---//
const Home = asyncLoad(() => import('./pages/home/Home'))
const ErrorPage = asyncLoad(() => import('./pages/errorPage/ErrorPage'));
const RefsPage = asyncLoad(() => import('./pages/refs/RefsPage'));
const ContextPage = asyncLoad(() => import('./pages/context/ContextPage'));

function App() {
  return (
    <BrowserRouter>
      <Switch>
        <Route path='/' component={Home} exact/>
        <Route path='/home' component={Home}/>
        <Route path='/err' component={ErrorPage}/>
        <Route path='/refs' component={RefsPage}/>
        <Route path='/context' component={ContextPage}/>
      </Switch>
    </BrowserRouter>
  );
}

export default App;
