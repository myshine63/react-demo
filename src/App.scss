.app {
  height: 100%;

  .aside-container {
    height: 100%;
    background-color: #3ba0e9;

    li {
      line-height: 40px;
      text-align: center;
      border-bottom: 1px solid white;

      a {
        color: white;
      }
    }
  }

  .ant-layout {
    background-color: white;
    padding: 16px;
  }
}