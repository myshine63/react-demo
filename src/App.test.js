import App from "./App";
import React from "react";
import renderer from 'react-test-renderer'

it('app组件', () => {
  const app = renderer.create(<App/>).toJSON();
  expect(app).toMatchSnapshot();
})