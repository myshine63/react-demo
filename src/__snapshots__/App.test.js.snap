// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app组件 1`] = `
<section
  className="ant-layout ant-layout-has-sider app"
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark"
    style={
      Object {
        "flex": "0 0 200px",
        "maxWidth": "200px",
        "minWidth": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <ul
        className="aside-container"
      >
        <li>
          <a
            href="/home"
            onClick={[Function]}
          >
            home
          </a>
        </li>
        <li>
          <a
            href="/err"
            onClick={[Function]}
          >
            err
          </a>
        </li>
        <li>
          <a
            href="/refs"
            onClick={[Function]}
          >
            refs
          </a>
        </li>
        <li>
          <a
            href="/router-page"
            onClick={[Function]}
          >
            router-page
          </a>
        </li>
        <li>
          <a
            href="/todo"
            onClick={[Function]}
          >
            todo
          </a>
        </li>
      </ul>
    </div>
  </aside>
  <section
    className="ant-layout"
  >
    <main
      className="ant-layout-content"
    >
      <div>
        loading
      </div>
    </main>
  </section>
</section>
`;
